trigger/trigger.go

```diff
@@ -278,10 +280,27 @@ func (t *triggerer) Trigger(ctx context.Context, repo *core.Repository, base *co
 		// TODO add instance
 		// TODO add target
 		// TODO add ref
+
+		// Insert Defaults
 		name := pipeline.Name
 		if name == "" {
 			name = "default"
 		}
+
+		// With the introduction of the "close" event in drone the default behaviour changes from "opt in to opens
+		// only" to "opt in to all".
+		//
+		// So as to preserve BC, this polyfills the previous design of "open" and "sync" events being included, but
+		// "exclude" events not. This allows users to "opt in" to the exclude behaviour.
+		if pipeline.Trigger.Event.Match(core.EventPullRequest) &&
+			len(pipeline.Trigger.Action.Include) == 0 {
```

Comment:
I would prefer we introduce a new skip function and add to the if / else block below, and then add unit tests to ensure we have coverage for all scenarios.

Reply:
I'm not quite sure what you mean. 

If you mean like `ifSkipPullRequest` or? My thinking is that's getting a little bespoke; following this conceptual model forward, there might be defaults that are changed as actions continue to be built on future.

Perhaps an alternative abstraction might be some functions that match a pipeline and fill defaults if there are otherwise none before they get passed to the matcher? Like, if you had a map of functions:

```go
// backfills/api.go
// Note: I am not familiar with this codebase; these examples may not match variable names or types directly
type DefaultBackfill func(pipline *Pipeline) 

var Backfills := []DefaultBackfill{PullRequestActions} // Can add backfills here in future
```

With a backfill impl like:

```go
func PullRequestActions(pipeline *Pipeline) {
    if ! pipeline.Trigger.Event.Match(core.EventPullRequest) {
      return
    }

    // Backfill logic here. 
}
```

That would be extensible and easily unit testable. 

Alternatively, can you explain what you mean by "new skip function"?

Reply:
> If you mean like ifSkipPullRequest or? My thinking is that's getting a little bespoke; following this conceptual model forward, there might be defaults that are changed as actions continue to be built on future.

yes, I mean adding an `ifSkipEventAction` which would skip pull request close events if the trigger does not explicitly enable this event and action.

Reply:
> The way this is currently implemented, in this pull request, is undesirable because it mutates the pipeline by modifying the trigger section. This is the sort of thing that comes back to bite me years later when I'm making changes and not expecting this to be mutated.

That makes sense. Okay, I'll reimplement it in this way. To set expectations up front, I don't know when I'll get this -- hopefully within a couple weeks, honestly i don't know.